### PR TITLE
Add: add more Biginteger methods

### PR DESCRIPTION
**Pull request by Liaojinghui**
This pr adds more BigInteger methods:

- Add
- Subtract
- Multiply
- Divide
- Negate
- Remainder
- Compare

Methods check list:

- [x] BigInteger.Add(BigInteger, BigInteger)
- [x] BigInteger.Compare(BigInteger, BigInteger)
- [x] BigInteger.Divide(BigInteger, BigInteger)
- [ ] BigInteger.DivRem(BigInteger, BigInteger, out BigInteger)
- [ ] BigInteger.Equals(object)
- [x] BigInteger.Equals(BigInteger)
- [x] BigInteger.Explicit(BigInteger to byte)
- [ ] BigInteger.Explicit(BigInteger to decimal)
- [ ] BigInteger.Explicit(BigInteger to double)
- [x] BigInteger.Explicit(BigInteger to short)
- [x] BigInteger.Explicit(BigInteger to int)
- [x] BigInteger.Explicit(BigInteger to long)
- [x] BigInteger.Explicit(BigInteger to sbyte)
- [ ] BigInteger.Explicit(BigInteger to float)
- [x] BigInteger.Explicit(BigInteger to ushort)
- [x] BigInteger.Explicit(BigInteger to uint)
- [x] BigInteger.Explicit(BigInteger to ulong)
- [x] BigInteger.GreatestCommonDivisor(BigInteger, BigInteger)
- [ ] BigInteger.GetHashCode()
- [ ] BigInteger.GetType()
- [x] BigInteger.Implicit(byte to BigInteger)
- [x] BigInteger.Implicit(decimal to BigInteger)
- [ ] BigInteger.Implicit(double to BigInteger)
- [x] BigInteger.Implicit(short to BigInteger)
- [x] BigInteger.Implicit(int to BigInteger)
- [x] BigInteger.Implicit(long to BigInteger)
- [x] BigInteger.Implicit(sbyte to BigInteger)
- [ ] BigInteger.Implicit(float to BigInteger)
- [x] BigInteger.Implicit(ushort to BigInteger)
- [x] BigInteger.Implicit(uint to BigInteger)
- [x] BigInteger.Implicit(ulong to BigInteger)
- [x] BigInteger.Max(BigInteger, BigInteger)
- [x] BigInteger.Min(BigInteger, BigInteger)
- [x] BigInteger.ModPow(BigInteger, BigInteger, BigInteger)
- [x] BigInteger.Multiply(BigInteger, BigInteger)
- [x] BigInteger.Negate(BigInteger)
- [ ] BigInteger.OnesComplement(BigInteger)
- [x] BigInteger.Parse(string)
- [ ] BigInteger.Parse(string, IFormatProvider)
- [ ] BigInteger.Parse(string, NumberStyles)
- [ ] BigInteger.Parse(string, NumberStyles, IFormatProvider)
- [x] BigInteger.Pow(BigInteger, int)
- [x] BigInteger.Remainder(BigInteger, BigInteger)
- [x] BigInteger.Subtract(BigInteger, BigInteger)
- [x] BigInteger.ToString()
- [ ] BigInteger.ToString(IFormatProvider)
- [ ] BigInteger.ToString(string)
- [ ] BigInteger.ToString(string, IFormatProvider)
- [ ] BigInteger.TryParse(string, out BigInteger)
- [ ] BigInteger.TryParse(string, NumberStyles, IFormatProvider, out BigInteger)

### CODE REVIEW

File: tests/Neo.Compiler.CSharp.UnitTests/TestClasses/Contract_BigInteger.cs [hunk 1]
```diff
@@ -123,5 +123,46 @@ public static bool testIsEven(BigInteger input)
...
+        {
+            return BigInteger.Compare(x, y);
+        }
+
+        public static BigInteger GreatestCommonDivisor(BigInteger x, BigInteger y)
+        {
+            return BigInteger.GreatestCommonDivisor(x, y);
+        }
+
     }
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/TestClasses/Contract_BigInteger.cs:167]:
```suggestion
    }
```

File: tests/Neo.Compiler.CSharp.UnitTests/UnitTest_BigInteger.cs [hunk 2]
```diff
@@ -304,5 +273,88 @@ public void Test_IsEven()
             Assert.AreEqual(new BigInteger(-2).IsEven, value);
             testengine.Reset();
         }
+
+
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/UnitTest_BigInteger.cs:277]:
```suggestion

```

File: tests/Neo.Compiler.CSharp.UnitTests/UnitTest_BigInteger.cs [hunk 3]
```diff
@@ -304,5 +272,86 @@ public void Test_IsEven()
             Assert.AreEqual(new BigInteger(-2).IsEven, value);
             testengine.Reset();
         }
+        [TestMethod]
+        public void Test_Add()
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/UnitTest_BigInteger.cs:276]:
```suggestion

        [TestMethod]
        public void Test_Add()
```

File: src/Neo.Compiler.CSharp/MethodConvert.cs [hunk 4]
```diff
@@ -4199,6 +4199,63 @@ private bool TryProcessSystemMethods(SemanticModel model, IMethodSymbol symbol,
...
+                    AddInstruction(OpCode.EQUAL);
+                    Jump(OpCode.JMPIF, gcdTarget);
+                    AddInstruction(OpCode.DUP);
+                    AddInstruction(OpCode.REVERSE3);
+                    AddInstruction(OpCode.SWAP);
+                    AddInstruction(OpCode.MOD);
+                    Jump(OpCode.JMP, whileTarget);
+                    gcdTarget.Instruction = AddInstruction(OpCode.NOP);
+                    AddInstruction(OpCode.DROP);
+                    AddInstruction(OpCode.ABS);
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/MethodConvert.cs:4257]: This test fail

File: tests/Neo.Compiler.CSharp.UnitTests/UnitTest_BigInteger.cs [hunk 5]
```diff
@@ -304,5 +272,87 @@ public void Test_IsEven()
...
+            Assert.AreEqual(BigInteger.Remainder(123456, 123), value);
+        }
+
+        [TestMethod]
+        public void Test_Compare()
+        {
+            testengine.Reset();
+            var result = testengine.ExecuteTestCaseStandard("testCompare", 123, 321);
+            var value = result.Pop().GetInteger();
+            Assert.AreEqual(BigInteger.Compare(123, 321), value);
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/UnitTest_BigInteger.cs:336]:
```suggestion
            testengine.Reset();
            var result = testengine.ExecuteTestCaseStandard("testCompare", 123, 321);
            var value = result.Pop().GetInteger();
            Assert.AreEqual(BigInteger.Compare(123, 321), value);

            testengine.Reset();
            result = testengine.ExecuteTestCaseStandard("testCompare", 123, 123);
            value = result.Pop().GetInteger();
            Assert.AreEqual(BigInteger.Compare(123, 123), value);

            testengine.Reset();
            result = testengine.ExecuteTestCaseStandard("testCompare", 123, -321);
            value = result.Pop().GetInteger();
            Assert.AreEqual(BigInteger.Compare(123, -321), value);
```

File: src/Neo.Compiler.CSharp/MethodConvert.cs (hunk 4 above)
Comment (Liaojinghui) [reply src/Neo.Compiler.CSharp/MethodConvert.cs:4257]: cause I actually tested compare,,,,,forgotten to change the test name

**Review by shargon — APPROVED** [review]

**Pull request merged**